node/src/components/consensus/protocols/highway.rs

```diff
@@ -403,11 +404,32 @@ where
                         )];
                     }
                 };
+                let is_faulty = match pvv.inner().signed_wire_unit() {
+                    Some(signed_wire_unit) => self
+                        .highway
+                        .state()
+                        .is_faulty(signed_wire_unit.wire_unit.creator),
+                    None => false,
+                };
+
                 match pvv.timestamp() {
                     Some(timestamp) if timestamp > Timestamp::now() => {
-                        self.store_vertex_for_addition_later(timestamp, sender, pvv)
+                        // If it's not from an equivocator and from the future, add to queue
+                        if !is_faulty {
+                            self.store_vertex_for_addition_later(timestamp, sender, pvv)
+                        } else {
+                            vec![]
+                        }
+                    }
+                    _ => {
+                        // If it's a transitive dependency or not from an equivocator, add the
+                        // vertex
+                        if self.vertex_deps.contains_key(&pvv.inner().id()) || !is_faulty {
```

Comment:
If you switch places and put `!is_faulty || …` first then this check should be quicker as I believe `||` won't check the second condition if first is `true`.

Reply:
Especially that we expect most votes be created by honest validators.